ui/ui-components/components/model/overview/SectionContainer.tsx

```diff
@@ -0,0 +1,49 @@
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Col, Media, Row } from "react-bootstrap";
+import AppIcon from "../../AppIcon";
+
+interface SectionContainerProps {
+  title: string;
+  description: string;
+  iconType?: "app" | "icon";
+  icon: string | IconProp;
+  actionButtons?: React.ReactNode;
+}
+
+const SectionContainer: React.FC<SectionContainerProps> = ({
+  actionButtons,
+  title,
+  iconType,
+  icon,
+  description,
+  children,
+}) => {
+  return (
+    <Media>
+      {icon && iconType === "app" ? (
```

Comment:
I think if you add `typeof icon === "string` in here you won't need the `as` below.

Reply:
Fancy